Solved crash rotating device and duplicated fragment in Splash

edwin-pixel-1 opened the pull request:
- Added support to save instance state in SubscriberList Fragment to avoid recreate the fields/attributes on configuration change like device rotation.
- Added support to save instance state in Splash activity to avoid duplicate fragment.

The next trello cards are solved:
- https://trello.com/c/tV40lonJ/16-crash-rotating-the-device
- https://trello.com/c/5BuUwqeu/11-rotating-the-device-the-splash-fragment-is-added-once-again

lucaslabs commented:
**No Merge** because this PR need to be splitted.

edwin-pixel-1 committed:
SaveInstanceState in SubscribersList and Splash

- Added support to use savedInstanceState in SubscribersListFragment class to avoid recreate everything on change configuration.

- Added support to savedInstanceState in Splash activity to avoid duplicate fragment on device rotation.

- Hide top bar in Splash

edwin-pixel-1 committed:
SaveInstanceState in SubscriberDetail

- Added support to use savedInstanceState in SubscriberDetailFragment class to avoid recreate everything on change configuration.

edwin-pixel-1 committed:
Merge branch 'master' into Resolution-Crash

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailFragment.java:
```diff
@@ -69,23 +74,24 @@ private void loadArgs() {
-        recyclerViewSubscribers.setLayoutManager(new LinearLayoutManager(getActivity()));
-        recyclerViewSubscribers.setAdapter(repositoriesAdapter);
-        repositoriesAdapter.setClickListener(this);
+        if (savedInstanceState == null) {
```
Use `if(getView() != null)` instead.

elabcius5 replied:
If we use this suggestion, the objects are recreated every time that a new fragment is created. The attributes instance aren't retained, they are new each time.

edwin-pixel-1 replied:
Yes, also at that point of lifeCycle of the fragment, the view is not been created yet.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersListFragment.java:
```diff
@@ -31,26 +31,34 @@
-        recyclerViewSubscribers.setLayoutManager(new LinearLayoutManager(getActivity()));
-        recyclerViewSubscribers.setAdapter(subscribersAdapter);
-        subscribersAdapter.setClickListener(this);
+        if (savedInstanceState == null) {
```
Use `if(getView() != null)` instead.

elabcius5 replied:
If we use this suggestion, the objects are recreated every time that a new fragment is created. The attributes instance aren't retained, they are new each time.

lucaslabs commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/view/SubscribersListFragment.java:
```diff
@@ -31,26 +31,34 @@
+            recyclerViewSubscribers.setLayoutManager(new LinearLayoutManager(getActivity()));
+            recyclerViewSubscribers.setAdapter(subscribersAdapter);
+            subscribersAdapter.setClickListener(this);
+        }else{
```
No need for this else.

edwin-pixel-1 committed:
Remake of save instance state strategy

- Added support to save Instance of SubscribersList in the presenter and
  recover it to solve the crash.

edwin-pixel-1 committed:
Fixed RecyclerView Row margin

- Fixed margin in row_repository.xml
- Fixed margin un row_subscriber.xml

edwin-pixel-1 committed:
Remake of save instance state strategy

- Added support to save Instance of SubscriberDetail fragment data and in
  the presenter data and recover it.

lucaslabs commented:
LG

daniellemariani commented on app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/interactor/SubscribersListInteractorImpl.java:
```diff
@@ -40,4 +40,11 @@ public void onFailure(Call<List<Subscriber>> call, Throwable t) {
+
+    @Override
+    public void onCancelRequest() {
+        if(call != null && call.isExecuted()){
```
Check code format

daniellemariani commented:
LG, just a minor comment

edwin-pixel-1 committed:
Fixed code format
